src/pretix/api/views/event.py

```diff
@@ -10,23 +13,89 @@
 from pretix.base.models.organizer import TeamAPIToken
 
 
-class EventViewSet(viewsets.ReadOnlyModelViewSet):
+class EventViewSet(viewsets.ModelViewSet):
     serializer_class = EventSerializer
     queryset = Event.objects.none()
     lookup_field = 'slug'
     lookup_url_kwarg = 'event'
+    permission_classes = (EventCRUDPermission,)
 
     def get_queryset(self):
         return self.request.organizer.events.prefetch_related('meta_values', 'meta_values__property')
 
+    def perform_update(self, serializer):
+        super().perform_update(serializer)
+        serializer.instance.log_action(
+            'pretix.event.changed',
+            user=self.request.user,
+            api_token=(self.request.auth if isinstance(self.request.auth, TeamAPIToken) else None),
+            data=self.request.data
+        )
+
+    def perform_create(self, serializer):
+        serializer.save(organizer=self.request.organizer)
+        serializer.instance.log_action(
+            'pretix.event.added',
+            user=self.request.user,
+            api_token=(self.request.auth if isinstance(self.request.auth, TeamAPIToken) else None),
+            data=self.request.data
+        )
+
+    def perform_destroy(self, instance):
+        if not instance.allow_delete():
+            raise PermissionDenied('The event can not be deleted as it already contains orders. Please set \'live\''
+                                   ' to false to hide the event and take the shop offline instead.')
+        try:
+            with transaction.atomic():
+                instance.organizer.log_action(
+                    'pretix.event.deleted', user=self.request.user,
+                    data={
+                        'event_id': instance.pk,
+                        'name': str(instance.name),
+                        'logentries': list(instance.logentry_set.values_list('pk', flat=True))
+                    }
+                )
+                instance.items.all().delete()
+                instance.subevents.all().delete()
+                super().perform_destroy(instance)
+        except ProtectedError:
+            raise PermissionDenied('The event could not be deleted as some constraints (e.g. data created by plug-ins) '
+                                   'do not allow it.')
+
+
+class CloneEventViewSet(viewsets.ModelViewSet):
+    serializer_class = EventSerializer
+    queryset = Event.objects.none()
+    lookup_field = 'slug'
+    lookup_url_kwarg = 'event'
+    http_method_names = ['post']
+    write_permission = 'can_create_events'
+
+    def perform_create(self, serializer):
+        serializer.save(organizer=self.request.organizer)
+        plugins = serializer.instance.plugins
+
+        event = Event.objects.filter(slug=self.kwargs['event'], organizer=self.request.organizer.pk).first()
+        serializer.instance.copy_data_from(event)
+
+        serializer.instance.plugins = plugins
```

Comment:
Hmm, do you think there would be a good idea to allow to specify that one wants to copy this as well? I'm not sure about a nice design here.

Reply:
I tried to consider what should determine this. The post body or the events that is cloned. I chose let the post body decide because I consider this the most explicit request. In the documentation I deliberately only wrote that "settings and items" are copied.

Reply:
I agree that it makes sense to honor the most explicit request. "Settings" is a vague term to many users, though, so we might make this *even more* explicit in the documentation.